File: HackerRank/Day8/main.js
function processData(input) {
    
    const lines = input.trim().split("\n");

    const n = parseInt(lines[0].trim());
    
    const phoneBook = new Map();
    
    
    for (let i = 1; i <= n; i++) {
        const [name, phone] = lines[i].trim().split(" ");
        phoneBook.set(name, phone);
    }
    
    for (let i = n + 1; i < lines.length; i++) {
        const queryName = lines[i].trim();
        
        if (!queryName) continue;
        
        if (phoneBook.has(queryName)) {
            console.log(`${queryName}=${phoneBook.get(queryName)}`);
        } else {
            console.log("Not found");
        }
    }
} 

process.stdin.resume();
process.stdin.setEncoding("ascii");
_input = "";
process.stdin.on("data", function (input) {
    _input += input;
});

process.stdin.on("end", function () {
   processData(_input);
});

File: HackerRank/Day8/main.cs
using System;
using System.Collections.Generic;
using System.IO;

class Solution {
    static void Main(String[] args) {
        
        int n = int.Parse(Console.ReadLine().Trim());
        
        Dictionary<string, string> phoneBook = new Dictionary<string, string>();
        
    
        for (int i = 0; i < n; i++) {
            string[] entry = Console.ReadLine().Split(' ');
            string name = entry[0];
            string phone = entry[1];
            
            phoneBook[name] = phone;
        }
        
        
        string query;
        while ((query = Console.ReadLine()) != null) {
            query = query.Trim();
            

            if (string.IsNullOrEmpty(query)) {
                break;
            }
            
            
            if (phoneBook.ContainsKey(query)) {
                Console.WriteLine($"{query}={phoneBook[query]}");
            } else {
                Console.WriteLine("Not found");
            }
        }
    }
}